downloader.py

```diff
@@ -25,91 +30,150 @@
 import argparse
 import glob
 
-# Default values
-create_folder = True
-create_plst = False
-
-# CMD arguments
-parser = argparse.ArgumentParser(
-    description='Download mp3 files from provided youtube playlist URL'
-)
-parser.add_argument(
-    '-u',
-    '--url',
-    action='store',
-    dest='url',
-    default=None,
-    help='Required youtube URL. You can pass either playlist or single video URL',
-    required=True
-)
-parser.add_argument(
-    '-f',
-    '--not-create-folder',
-    action='store_false',
-    dest='fold',
-    help='Don\'t create a folder'
-)
-parser.add_argument(
-    '-p',
-    '--playlist',
-    action='store_true',
-    dest='plst',
-    help='Create an m3u playlist from downloaded files'
-)
-
-parser.set_defaults(fold=True, plst=False)
-results = parser.parse_args()
-url = results.url
-create_folder = results.fold
-create_plst = results.plst
-
-# Get playlist title and create a folder for it the argument
-# -f wasn't present
-foldername = ''
-if create_folder:
-    ydl_init_opts = {
-        'quiet': True,
-        'extract_flat': True
+
+def dl_audio(url, folder=True, create_plst=False, download_plst=True):
+    """Download audio from provided url and save it to hard drive."""
+    # Get playlist title and create a folder for it the argument
+    # -f wasn't present
+    foldername = ''
+    if folder:
+        if isinstance(url, list):
+            # the last item in the list is filename
```

Comment:
Why is "the last item is file name"? 
We don't need the file name itself, we only use it to decide what should be the folder name, right?

If so, then I think that we should pass the folder name itself as an argument and always require the first parameter to be the list:

```
dl_audio(urls, folder=None, create_plst=False, download_plst=True)
```

Here if `folder` is `None` we don't create a folder (use current folder) and if it's not `None` - it is a folder name to create.